collated/main/lawwman.md

```diff
@@ -0,0 +1,888 @@
+     */
+    @Override
+    public synchronized ReadOnlyPerson removeOverdueDebtPerson(ReadOnlyPerson target) throws PersonNotFoundException {
+        ReadOnlyPerson overdueDebtPerson = addressBook.removeOverdueDebtPerson(target);
+        updateFilteredOverduePersonList(PREDICATE_SHOW_ALL_OVERDUE_PERSONS);
+        indicateAddressBookChanged();
+        return overdueDebtPerson;
+    }
+
+```
+###### \java\seedu\address\model\ModelManager.java
+``` java
+    /**
+     * Adds a specific person to overdue list in the AddressBook.
+     * @param person to be updated.
+     * @return overdueDebtPerson
+     * @throws DuplicatePersonException if this operation causes a contact to be a duplicate of another.
+     */
+    @Override
+    public synchronized ReadOnlyPerson addOverdueDebtPerson(ReadOnlyPerson person) {
+        ReadOnlyPerson overdueDebtPerson = addressBook.addOverdueDebtPerson(person);
+        updateFilteredOverduePersonList(PREDICATE_SHOW_ALL_OVERDUE_PERSONS);
+        indicateAddressBookChanged();
+        return overdueDebtPerson;
+    }
+
+```
+###### \java\seedu\address\model\ModelManager.java
+``` java
+    /**
+     * Reads the masterlist and updates the overdue list accordingly.
+     */
+    public void syncOverdueList() {
+        filteredOverduePersons = new FilteredList<>(this.addressBook.getOverduePersonList());
+    }
+
+```
+###### \java\seedu\address\model\ModelManager.java
+``` java
+    /**
+     * Returns an unmodifiable view of the overdue list of {@code ReadOnlyPerson} backed by the internal list of
+     * {@code addressBook}
+     */
+    @Override
+    public ObservableList<ReadOnlyPerson> getFilteredOverduePersonList() {
+        setCurrentListName("overduelist");
+        syncOverdueList();
+        filteredOverduePersons.setPredicate(currentPredicate);
+        return FXCollections.unmodifiableObservableList(filteredOverduePersons);
+    }
+
+```
+###### \java\seedu\address\model\ModelManager.java
+``` java
+    /**
+     * Filters {@code filteredOverduePersons} according to given {@param predicate}
+     * @return size of current displayed filtered list.
+     */
+    @Override
+    public int updateFilteredOverduePersonList(Predicate<ReadOnlyPerson> predicate) {
+        requireNonNull(predicate);
+        currentPredicate = predicate;
+        filteredOverduePersons.setPredicate(predicate);
+        return filteredOverduePersons.size();
+    }
+
+```
+###### \java\seedu\address\model\ModelManager.java
+``` java
+    @Subscribe
+    public void handleLoginUpdateDebt(LoginAppRequestEvent event) {
+        // login is successful
+        if (event.getLoginStatus() == true) {
+            for (ReadOnlyPerson person : allPersons) {
+                if (!person.getInterest().value.equals("No interest set.")
+                        && (person.checkLastAccruedDate(new Date()) != 0)) {
+                    updateDebtFromInterest(person, person.checkLastAccruedDate(new Date()));
```

Comment:
Avoid arrow head code. Perhaps abstracting away the for statement at line 226 into a method improve this.